extensions/podman/packages/extension/src/platforms/win-platform.ts

```diff
@@ -0,0 +1,88 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { ExtensionContext, InstallCheck, TelemetryLogger } from '@podman-desktop/api';
+import * as extensionApi from '@podman-desktop/api';
+
+import { OrCheck, SequenceCheck } from '../checks/base-check';
+import { HyperVCheck } from '../checks/hyperv-check';
+import { HyperVPodmanVersionCheck } from '../checks/hyperv-podman-version-check';
+import { VirtualMachinePlatformCheck } from '../checks/virtual-machine-platform-check';
+import { WinBitCheck } from '../checks/win-bit-check';
+import { WinMemoryCheck } from '../checks/win-memory-check';
+import { WinVersionCheck } from '../checks/win-version-check';
+import { WSLVersionCheck } from '../checks/wsl-version-check';
+import { WSL2Check } from '../checks/wsl2-check';
+
+export class WinPlatform {
+  readonly type = 'win';
+
+  private readonly winBitCheck: WinBitCheck;
+  private readonly winVersionCheck: WinVersionCheck;
+  private readonly winMemoryCheck: WinMemoryCheck;
+  private readonly virtualMachinePlatformCheck: VirtualMachinePlatformCheck;
+  private readonly wslVersionCheck: WSLVersionCheck;
+  private readonly wsl2Check: WSL2Check;
+  private readonly hyperVCheck: HyperVCheck;
+  private readonly hyperVPodmanVersionCheck: HyperVPodmanVersionCheck;
+
+  constructor(
+    private extensionContext: ExtensionContext,
+    private telemetryLogger: TelemetryLogger,
+  ) {
+    this.winBitCheck = new WinBitCheck();
+    this.winVersionCheck = new WinVersionCheck();
+    this.winMemoryCheck = new WinMemoryCheck();
+    this.virtualMachinePlatformCheck = new VirtualMachinePlatformCheck(this.telemetryLogger);
+    this.wslVersionCheck = new WSLVersionCheck();
+    this.wsl2Check = new WSL2Check(this.telemetryLogger, this.extensionContext);
+    this.hyperVCheck = new HyperVCheck(this.telemetryLogger);
+    this.hyperVPodmanVersionCheck = new HyperVPodmanVersionCheck();
+  }
+
+  getPreflightChecks(): InstallCheck[] {
+    return [
+      this.winBitCheck,
+      this.winVersionCheck,
+      this.winMemoryCheck,
+      new OrCheck(
+        'Windows virtualization',
+        new SequenceCheck('WSL platform', [this.virtualMachinePlatformCheck, this.wslVersionCheck, this.wsl2Check]),
+        this.hyperVCheck,
+      ),
+    ];
+  }
+
+  async isWSLEnabled(): Promise<boolean> {
+    if (!extensionApi.env.isWindows) {
+      return false;
+    }
+    const wslCheck = new SequenceCheck('WSL platform', [this.wslVersionCheck, this.wsl2Check]);
```

Comment:
We could probably simplify the private property of the class, by directly registering sequence check for WSL check, like avoiding to re-create sequence and or checks